prime-router/src/main/kotlin/azure/CheckFunction.kt

```diff
@@ -107,19 +134,46 @@ class CheckFunction : Logging {
     /**
      * Any normal return is success.  Any exception thrown is failure.
      */
-    fun testSftp(sftpTransportType: SFTPTransportType, receiver: Receiver, responseBody: MutableList<String>) {
+    fun testSftp(sftpTransportType: SFTPTransportType, receiver: Receiver, sftpFile: SftpFile?, responseBody: MutableList<String>) {
         val host = sftpTransportType.host
         val port = sftpTransportType.port
         val path = sftpTransportType.filePath
+        logger.info("SFTP Transport $sftpTransportType")
+        responseBody.add("${receiver.fullName}: SFTP Transport: $sftpTransportType")
         val credential = SftpTransport.lookupCredentials(receiver.fullName)
-        val sshClient = SftpTransport.connect(host, port, credential)
-        responseBody.add("${receiver.fullName}: Able to Connect to sftp site.  Now trying an `ls`...")
+        var sshClient = SftpTransport.connect(host, port, credential)
+        responseBody.add("${receiver.fullName}: Able to Connect to sftp site")
+        sftpFile?. let {
+            logger.info("Attempting to upload ${it.name} to $sftpTransportType")
+            SftpTransport.uploadFile(sshClient, path, it.name, it.contents.toByteArray())
+            responseBody.add("${receiver.fullName}: Uploaded file '${sftpFile.name}' to SFTP transport")
+            // the client connection is closed in the SftpTransport methods
+            sshClient = SftpTransport.connect(host, port, credential)
+        }
+        logger.info("Now trying an `ls` on $path")
         val lsList: List<String> = SftpTransport.ls(sshClient, path)
         // Log what we found from ls, but don't return it.
         logger.info("What we got back from ls (first few lines): ")
         lsList.filterIndexed { index, _ -> index <= 5 }.forEach { logger.info(it) }
-        val msg = "${receiver.fullName}: Success: ls returned ${lsList.size} rows of info from $sftpTransportType"
+        var msg = "${receiver.fullName}: Success: ls returned ${lsList.size} rows of info from SFTP Transport"
         logger.info(msg)
         responseBody.add(msg)
+        sftpFile?. let {
+            logger.info("Checking for uploaded file in `ls` results")
+            msg = if (lsList.filter { it.endsWith("$path/${sftpFile.name}")}.isEmpty()) {
+                "${receiver.fullName}: Couldn't find file '${sftpFile.name}' in `ls` results: ${lsList.toString()}"
+            } else {
+                "${receiver.fullName}: Found uploaded file '${sftpFile.name}' in `ls` results"
+            }
+            logger.info(msg)
+            responseBody.add(msg)
+            msg = "${receiver.fullName}: Removing '${sftpFile.name}' from SFTP transport"
+            logger.info(msg)
+            responseBody.add(msg)
+            SftpTransport.rm(SftpTransport.connect(host, port, credential), path, sftpFile.name)
+            msg = "${receiver.fullName}: Success: removed '${sftpFile.name}' from SFTP Transport"
```

Comment:
I like that we're cleaning up our own record, but given the possibility that we could have a name collision and then remove or overwrite an existing file, I'd like to skip the `rm` if there was a collision by throwing an exception or branching out above.

Reply:
Last commit checks to see if the file exists and will throw an exception before the upload.